From b20c8177092f3ff4804e0271e3fd338d34e09097 Mon Sep 17 00:00:00 2001
From: "Markus Schneider-Pargmann (The Capable Hub)" <msp@baylibre.com>
Date: Tue, 28 Apr 2026 10:30:29 +0200
Subject: [PATCH 04439/15333] tracing: fprobe: Remove __packed from generic
 __fprobe_header

fp pointer and unsigned long have the same size on all relevant
architectures that build Linux. Furthermore this struct is only used in
architectures that do not set ARCH_DEFINE_ENCODE_FPROBE_HEADER which is
set only for 64bit architectures (apart from LoongArch).

Both fields are aligned on these architectures so the struct with
__packed and without it are the same.

Remove the __packed as it is unnecessary.

[Masami: Removed Fixes tag because this is not fixing any problem.]

Link: https://lore.kernel.org/all/20260428-topic-fprobe-packed-v7-1-v1-1-9abc9b866b4c@baylibre.com/

Signed-off-by: Markus Schneider-Pargmann (The Capable Hub) <msp@baylibre.com>
Signed-off-by: Masami Hiramatsu (Google) <mhiramat@kernel.org>
---
 kernel/trace/fprobe.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/trace/fprobe.c b/kernel/trace/fprobe.c
index f215990b9061..2727f8861fa3 100644
--- a/kernel/trace/fprobe.c
+++ b/kernel/trace/fprobe.c
@@ -181,7 +181,7 @@ static inline void read_fprobe_header(unsigned long *stack,
 struct __fprobe_header {
 	struct fprobe *fp;
 	unsigned long size_words;
-} __packed;
+};
 
 #define FPROBE_HEADER_SIZE_IN_LONG	SIZE_IN_LONG(sizeof(struct __fprobe_header))
 
-- 
2.55.0

